test/requests/diagnostics_expectations_test.rb

```diff
@@ -15,13 +15,12 @@ def run_expectations(source)
 
     document = RubyLsp::Document.new(source: source, version: 1, uri: "file://#{__FILE__}")
     RubyLsp::Requests::Diagnostics.new(document).run
-    result = T.let(nil, T.nilable(T::Array[RubyLsp::Requests::Support::RuboCopDiagnostic]))
+    result = T.let([], T::Array[RubyLsp::Requests::Support::RuboCopDiagnostic])
 
     stdout, _ = capture_io do
-      result = T.cast(
-        RubyLsp::Requests::Diagnostics.new(document).run,
-        T::Array[RubyLsp::Requests::Support::RuboCopDiagnostic],
-      )
+      diagnostics = []
+      diagnostics.push(RubyLsp::Requests::Diagnostics.new(document).run)
+      result = diagnostics.compact.flatten
```

Comment:
What error is this change fixing? Aren't we able to just do `result = RubyLsp::Requests::Diagnostics.new(document).run`?

Reply:
If the document doesn't have valid syntax, the request return early, and this is raising an error below due to `T.must()`
Some fixtures aren't considered as valid on older ruby version, for example: https://github.com/ruby/yarp/blob/main/test/fixtures/whitequark/forwarded_kwrestarg_with_additional_kwarg.txt.